controllers/nnf_workflow_controller_helpers.go

```diff
@@ -1243,36 +1243,107 @@ func (r *NnfWorkflowReconciler) waitForContainersToStart(ctx context.Context, wo
+	// Retrieve the profile to extract the PostRun timeout
+	profile, err := r.getContainerProfile(ctx, workflow, index)
+	if err != nil {
+		return nil, err
+	}
+	timeout := time.Duration(profile.Data.PostRunTimeoutSeconds) * time.Second
 
 	// Ensure all the jobs are done running before we check the conditions.
 	for _, job := range jobList.Items {
+		// Jobs will have conditions when finished
 		if len(job.Status.Conditions) <= 0 {
-			return false, false, nil
+
+			// If desired, set the ActiveDeadline on the job to kill pods. Use the job's creation
+			// timestamp to determine how long the job/pod has been running at this point. Then, add
+			// the desired timeout to that value. k8s Job's ActiveDeadLineSeconds will then
+			// terminate the pods once the deadline is hit.
+			if timeout > 0 && job.Spec.ActiveDeadlineSeconds == nil {
+				deadline := int64((metav1.Now().Sub(job.CreationTimestamp.Time) + timeout).Seconds())
+
+				// Update the job with the deadline
+				err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+					j := &batchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: job.Name, Namespace: job.Namespace}}
+					if err := r.Get(ctx, client.ObjectKeyFromObject(j), j); err != nil {
+						return client.IgnoreNotFound(err)
+					}
+
+					j.Spec.ActiveDeadlineSeconds = &deadline
+					return r.Update(ctx, j)
+				})
+
+				if err != nil {
+					return nil, nnfv1alpha1.NewWorkflowErrorf("error updating job '%s' activeDeadlineSeconds:", job.Name)
+				}
+			}
+
+			return Requeue("pending container finish").after(2 * time.Second).withObject(&job), nil
 		}
 	}
 
-	// Once we know everything is done, we can check the conditions
+	return nil, nil
+}
+
+func (r *NnfWorkflowReconciler) checkContainersResults(ctx context.Context, workflow *dwsv1alpha1.Workflow, index int) (*result, error) {
+	jobList, err := r.getContainerJobs(ctx, workflow, index)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(jobList.Items) < 1 {
+		return nil, nnfv1alpha1.NewWorkflowErrorf("no container jobs found for workflow %s", workflow.Name)
+	}
+
 	for _, job := range jobList.Items {
 		for _, condition := range job.Status.Conditions {
 			if condition.Type != batchv1.JobComplete {
-				return true, false, nnfv1alpha1.NewWorkflowErrorf("container job %s (%s): %s", condition.Type, condition.Reason, condition.Message)
+				return nil, nnfv1alpha1.NewWorkflowErrorf("container job %s (%s): %s", condition.Type, condition.Reason, condition.Message)
 			}
 		}
 	}
 
-	// TODO: Once postrun starts, we will want a timeout that starts and eventually kills/fails job pods
-	// once that timeout is hit.
+	return nil, nil
+}
+
+func (r *NnfWorkflowReconciler) getContainerJobs(ctx context.Context, workflow *dwsv1alpha1.Workflow, index int) (*batchv1.JobList, error) {
+	jobList := &batchv1.JobList{}
```

Comment:
Precedence has the lables (the configurable stuff) above the returned object `jobList`, above the r.List() method

So
```go
matchLabels := dwsv1alpha1.MatchingWorkflow(workflow)
matchLabels[nnfv1alpha1.DirectiveIndexLabel] = strconv.Itoa(index)

jobs := &batchv1.JobList{}
if err := r.List(ctx, jobList, matchLabels); err != nil {
   ...
}
```

Reply:
Everywhere else we've wrapped matchLabels in a []client.ListOption array, but yours makes more sense. Might be worth a cleanup ticket sometime.